### DISCUSSION TITLE
Multiple domains feature: How to avoid running same slugs from unassigned domains?

### DISCUSSION THREAD

**Discussion by morphinestyle** [shlinkio/shlink#2004] [Help wanted] [answered]
### Shlink version

3.7.3

### PHP version

8.1

### How do you serve Shlink

Other (explain in summary)

### Database engine

MariaDB

### Database version

10.6

### Summary

Using the multiple domains feature, how can I prevent a slug assigned to the aaa.com domain from being recalled from the bbb.com domain?

For example: my short link is `aaa.com/test`, the problem is that it also works by entering the url `bbb.com/test`.
How can it be avoided?

**Comment by acelaya** [comment]
For backwards compatibility reasons, all short URLs will work with the default domain, but not with any non-default one.

The best option here is that you never use the default domain if you want to prevent this.

You can find more information about how Shlink handles multiple domains here https://shlink.io/documentation/advanced/multiple-domains/#visits

**Reply by morphinestyle** [reply]
The problem is that it also happens with domains that are not the default one.

Continuing the example from the previous post, my default domain would be zzz.com, and aaa.com/test and bbb.com/test lead to the same page.

How can I solve it?

**Reply by acelaya** [reply]
Can you provide a reproducible example, or some steps to reproduce it?

**Reply by acelaya** [reply]
I have quickly tested this, and I can't reproduce it in normal circumstances.

Steps I followed:

* Pointed aaa.com and bbb.com to my Shlink instance
* Created a new short URL, with custom slug aaa and domain aaa.com
* Tried to visit aaa.com/aaa -> It redirected as expected
* Created a new short URL, with custom slug bbb and domain bbb.com
* Tried to visit bbb.com/bbb -> It redirected as expected
* Tried to visit aaa.com/bbb -> 404, invalid short URL
* Tried to visit bbb.com/aaa -> 404, invalid short URL

When you provide the reproducible example, make sure to describe how do you serve Shlink (you skippedthis in the description) and if you are using something like a reverse proxy.

A reverse proxy might be causing the domain to not be properly propagated to Shlink. One way to check this is visiting an invalid short URL in one of the non-default domains, then going to the orphan visits, where you can see the actual visited URL, and verify that it was tracked with the actual non-default domain.

**Reply by morphinestyle** [reply, marked as answer]
I don't know the reason Alejandro but today it is working correctly without having changed anything.

The only cause that could have made Shlink work that way could be that Cyberpanel updates and restarts the webserver every 3 minutes (which in turn loads the updated .htaccess), and I ran those tests as soon as I added the domains.

I have no other hypotheses at the moment.
In any case: better this way, problem solved.
